src/WebJobs.Extensions.DurableTask.Scale/AzureStorage/AzureStorageScalabilityProvider.cs

```diff
@@ -0,0 +1,95 @@
+{
+    /// <summary>
+    /// Azure Storage backend implementation of the scalability provider for Durable Functions scaling decisions.
+    /// </summary>
+    public class AzureStorageScalabilityProvider : ScalabilityProvider
+    {
+        private readonly StorageAccountClientProvider storageAccountClientProvider;
+        private readonly string connectionName;
+        private readonly ILogger logger;
+
+        private readonly object initLock = new object();
+
+        private DurableTaskMetricsProvider singletonDurableTaskMetricsProvider;
+
+        public AzureStorageScalabilityProvider(
+            StorageAccountClientProvider storageAccountClientProvider,
+            string connectionName,
+            ILogger logger)
+            : base("AzureStorage", connectionName)
+        {
+            this.storageAccountClientProvider = storageAccountClientProvider ?? throw new ArgumentNullException(nameof(storageAccountClientProvider));
+            this.connectionName = connectionName;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// The app setting containing the Azure Storage connection string.
+        /// </summary>
+        public override string ConnectionName => this.connectionName;
+
+        internal DurableTaskMetricsProvider GetMetricsProvider(
+            string hubName,
+            StorageAccountClientProvider storageAccountClientProvider,
+            ILogger logger)
+        {
+            return new DurableTaskMetricsProvider(hubName, logger, performanceMonitor: null, storageAccountClientProvider);
+        }
+
+        /// <inheritdoc/>
+        public override bool TryGetScaleMonitor(
+            string functionId,
+            string functionName,
+            string hubName,
+            string connectionName,
+            out IScaleMonitor scaleMonitor)
+        {
+            lock (this.initLock)
+            {
+                if (this.singletonDurableTaskMetricsProvider == null)
+                {
+                    this.singletonDurableTaskMetricsProvider = this.GetMetricsProvider(
+                        hubName,
+                        this.storageAccountClientProvider,
+                        this.logger);
+                }
+
+                scaleMonitor = new DurableTaskScaleMonitor(functionId, hubName, this.logger, this.singletonDurableTaskMetricsProvider);
+                return true;
+            }
+        }
+
+        public override bool TryGetTargetScaler(
+            string functionId,
+            string functionName,
+            string hubName,
+            string connectionName,
+            out ITargetScaler targetScaler)
+        {
+            lock (this.initLock)
+            {
+                if (this.singletonDurableTaskMetricsProvider == null)
+                {
+                    // This is only called by the ScaleController, it doesn't run in the Functions Host process.
```

Comment:
This comment is true for all code in this new assembly, right? If so, we should delete this comment.